### PR TITLE
ci: lint all the things

### PR DESCRIPTION
**Pull request by dabeeeenster**
This does a few things - has to be in one PR otherwise builds will fail

- Moves all CI linting out of individual actions and puts them in their own action
- Lints everything that is left

### CODE REVIEW

**Comment by dabeeeenster** [comment]
@kyle-ssg I need some checks on this commit https://github.com/Flagsmith/flagsmith/pull/2446/commits/68740a2ef26845fdaf2f0d0b7eab4da325ec6ce7 also need to go through some of the files excluded:

```
frontend/web/components/tags/TagValues.tsx
frontend/web/components/pages/EnvironmentSettingsPage.js
frontend/common/utils/base/_utils.js
``` 

in `eslintignore`

File: .eslintignore [hunk 1]
```diff
@@ -7,4 +7,7 @@ frontend/web/static
 frontend/web/project/libs.js
 frontend/e2e
 frontend/webpack
-frontend/api
\ No newline at end of file
+frontend/api
+frontend/web/components/tags/TagValues.tsx
+frontend/web/components/pages/EnvironmentSettingsPage.js
+frontend/common/utils/base/_utils.js
```
Comment (matthewelwell) [inline .eslintignore]: is there any explanation why these are ignored?

**Review by matthewelwell — COMMENTED** [review]
One thing that I am concerned about here (but maybe can be resolved in the makefile) is that it's kinda painful to actually apply linting changes. 

Previously, if I needed to make some linting changes on multiple files, I would just run `black . && isort .`. We now will be left with `make lint` which passes the `--check` (or similar) argument.

File: .eslintignore (hunk 1 above)
Comment (dabeeeenster) [reply .eslintignore]: My lack of JS. @kyle-ssg might be able to fix these?

**Comment by dabeeeenster** [comment]
We can have `make lint-fix` or similar? 

`pre-commit run --all-files` or `pre-commit run --all-files <linter>` achieves this

**Pull request closed without merging**